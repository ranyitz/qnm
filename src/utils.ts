export function isTruthy<T>(x: T | undefined | null): x is T {
  return x !== undefined && x !== null;
}
