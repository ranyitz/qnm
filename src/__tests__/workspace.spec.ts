import Workspace from '../workspace/workspace';
import { resolveFixture } from './utils';

describe('workspace', () => {
  it('should throw an error for no package case', () => {
    try {
      Workspace.loadSync('/no/package/case/@@/__');
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it('should throw an error for no node_modules case', () => {
    try {
      Workspace.loadSync(resolveFixture('workspace/no-node_modules'));
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it('should throw an error for empty node_modules case', () => {
    try {
      Workspace.loadSync(resolveFixture('workspace/empty-node_modules'));
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });
});
