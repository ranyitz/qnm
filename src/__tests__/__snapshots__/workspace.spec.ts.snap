// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workspace should throw an error for no node_modules case 1`] = `[Error: could not find node_modules directory]`;

exports[`workspace should throw an error for no package case 1`] = `[Error: could not identify package directory]`;
